### PR TITLE
Workaround for #19883: Check if applied locale correctly lowers chars and fallback

### PR DESCRIPTION
**Pull request by basilgello**
## Description

Matrix-only workaround for #19883 . We agreed that master has to get a proper fix.

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by basilgello** [comment]
@lrusak

**Comment by basilgello** [comment]
I think we need it in 19.2 :)

**Comment by ksooo** [comment]
> I think we need it in 19.2 :)

 Then, get it into master first. No merges of backports for Matrix if not in master.

File: tools/Linux/kodi.sh.in [hunk 1]
```diff
@@ -171,6 +171,11 @@ if command_exists gdb; then
   fi
 fi
 
+
+# Check if locale is affected by "Turkish I"
+# See https://github.com/xbmc/xbmc/issue/19883 for details
```
Comment (bugwelle) [inline tools/Linux/kodi.sh.in:176]:
```suggestion
# See https://github.com/xbmc/xbmc/issues/19883 for details
```

**Comment by ksooo** [comment]
@basilgello a suggestion I have. Instead of having to sync your backport PR again and again, please mark the PR as WIP, get approval for your master PR and once you have it do the final sync for your backport PR. Saves you some time. ;-)

**Comment by basilgello** [comment]
> Instead of having to sync your backport PR again and again, please mark the PR as WIP

Yes, I am just not used to labelling yet. All in all, it is my first day as full team member :)

**Comment by ksooo** [comment]
As we've just discussed in the master PR, this workaround should be fine for Matrix. I will remove the WIP label and approve and merge it asap, okay?

File: tools/Linux/kodi.sh.in (hunk 1 above)
Comment (ksooo) [reply tools/Linux/kodi.sh.in:176]: good catch.

**Comment by basilgello** [comment]
> As we've just discussed in the master PR, this workaround should be fine for Matrix. I will remove the WIP label and approve and merge it asap, okay?

Yes, please!

**Comment by ksooo** [comment]
Once you have addressed the comment from @bugwelle ;-)

File: xbmc/LangInfo.cpp [hunk 2]
```diff
@@ -979,6 +979,18 @@ void CLangInfo::SetCurrentRegion(const std::string& strName)
 
   m_currentRegion->SetGlobalLocale();
 
+  // Check if locale is affected by "Turkish I"
+  // See https://github.com/xbmc/xbmc/issue/19883 for details
```
Comment (ksooo) [inline xbmc/LangInfo.cpp:983]: link does not work issue -> issues

Comment (basilgello) [reply xbmc/LangInfo.cpp:983]: Moment :)

Comment (basilgello) [reply xbmc/LangInfo.cpp]: Done, and incorporated code fixes

**Comment by ksooo** [comment]
Now, please fix the code style issue and we are ready to go.

**Comment by basilgello** [comment]
> Now, please fix the code style issue and we are ready to go.

I have applied it as third commit!

**Comment by ksooo** [comment]
> I have applied it as third commit!

 Please squash.

**Comment by basilgello** [comment]
> > I have applied it as third commit!
> 
> Please squash.

Done! :)

**Review by ksooo — APPROVED** [review]

**Review by howie-f — APPROVED** [review]
👍

**Comment by ksooo** [comment]
jenkins build this please

**Pull request merged**